gui/default/syncthing/core/syncthingController.js

```diff
@@ -210,6 +210,10 @@ angular.module('syncthing.core')
                 if (data.from === 'scanning' && data.to === 'idle') {
                     refreshFolderStats();
                 }
+
+                $scope.folders[data.folder].devices.forEach(function (deviceCfg) {
+                    refreshCompletion(deviceCfg.deviceID, data.folder);
+                });
```

Comment:
We get this data already in the FolderCompletion events, which is more efficient than asking to recalculate it here. Can we use that data instead?

Reply:
A folder rescan did not emit FolderCompletion event, so the sync status would not immediately update. I realize now this would be better placed under local index change event (when actual changes occur)

Reply:
The folder completion event is triggered (internally) by a folder changing state, though there may be a delay.

Reply:
Ah I see, disconnected devices are skipped